base/loading.jl

```diff
@@ -1653,6 +1656,7 @@ const include_callbacks = Any[]
 
 # used to optionally track dependencies when requiring a module:
 const _concrete_dependencies = Pair{PkgId,UInt128}[] # these dependency versions are "set in stone", and the process should try to avoid invalidating them
+const _mandatory_dependencies = Pair{PkgId,UInt128}[] # Like `_concrete_dependencies`, but required to actually be loaded (in order) in every precompile process
```

Comment:
Is this needed since the `--check-bounds=no` stdlib is a "virtual dependency"?
Could we avoid duplicating this logic by having the load happen in Julia proper?

I am not fully sure why the `__precompile__(:mandatory)` is needed, so expanding comments throughout the code would be helpful.

Reply:
I don't understand the question.